### ISSUE TITLE
possible bug in combiner

### ISSUE THREAD

**Issue by orialb** [ITensor/ITensors.jl#395]
The following code:
```julia
i1 = Index([QN(0)=>1,QN(1)=>2],"i1")
i2 = Index([QN(0)=>1,QN(1)=>2],"i2")
A = randomITensor(QN(),i1,i2,dag(i1)', dag(i2)')
CL = combiner(i1,i2)
CR = combiner(dag(i1)',dag(i2)')
AC = A *CR *CL

AC*dag(CR)*dag(CL)
```

results in an error:
```
ERROR: BoundsError: attempt to access Dim 1: 1
Dim 2: 1
Dim 3: 4
NDTensors.Dense{Float64,SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true}}
 1×1×4
 at index [1, 2, 3, 4, 5, 6, 7, 8]
Stacktrace:
 [1] copyto!(::IndexLinear, ::NDTensors.Tensor{Float64,3,NDTensors.Dense{Float64,SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true}},Tuple{Int
64,Int64,Int64}}, ::IndexCartesian, ::SubArray{Float64,2,Base.ReshapedArray{Float64,2,SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true},Tuple
{}},Tuple{UnitRange{Int64},UnitRange{Int64}},false}) at ./abstractarray.jl:813
 [2] copyto! at ./abstractarray.jl:798 [inlined]
 [3] uncombine(::NDTensors.Tensor{Float64,2,NDTensors.BlockSparse{Float64,Array{Float64,1},2},IndexSet{2,Index{Array{Pair{QN,Int64},1}},Tuple{Index{Array{Pai
r{QN,Int64},1}},Index{Array{Pair{QN,Int64},1}}}}}, ::IndexSet{3,Index{Array{Pair{QN,Int64},1}},Tuple{Index{Array{Pair{QN,Int64},1}},Index{Array{Pair{QN,Int64
},1}},Index{Array{Pair{QN,Int64},1}}}}, ::Int64, ::Array{Int64,1}, ::Array{Int64,1}) at /Users/oria/src/NDTensors.jl/src/blocksparse/blocksparsetensor.jl:621
 [4] contract(::NDTensors.Tensor{Float64,2,NDTensors.BlockSparse{Float64,Array{Float64,1},2},IndexSet{2,Index{Array{Pair{QN,Int64},1}},Tuple{Index{Array{Pair
{QN,Int64},1}},Index{Array{Pair{QN,Int64},1}}}}}, ::Tuple{Int64,Int64}, ::NDTensors.Tensor{Number,3,NDTensors.Combiner,IndexSet{3,Index{Array{Pair{QN,Int64},
1}},Tuple{Index{Array{Pair{QN,Int64},1}},Index{Array{Pair{QN,Int64},1}},Index{Array{Pair{QN,Int64},1}}}}}, ::Tuple{Int64,Int64,Int64}) at /Users/user/src/NDT
ensors.jl/src/blocksparse/combiner.jl:40
 [5] *(::ITensor{2}, ::ITensor{3}) at /Users/user/src/ITensors.jl/src/itensor.jl:1028
 [6] top-level scope at REPL[57]:1
 [7] eval(::Module, ::Any) at ./boot.jl:331
 [8] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [9] run_backend(::REPL.REPLBackend) at /Users/user/.julia/packages/Revise/MgvIv/src/Revise.jl:1023
 [10] top-level scope at REPL[2]:0
```

If I call `AC*dag(CL)*dag(CR)` instead,  it works.
I am assuming that the order shouldn't really matter here and this is a bug, but maybe I'm doing something wrong?

**Comment by mtfishman** [comment]
Wow, I'm surprised this bug slipped through, we have a lot of combiner tests. Thanks for pointing it out. Looks like it already errors with just `AC * dag(CR)`. I'll take a look, hopefully it won't be hard to fix.

**Comment by mtfishman** [comment]
This was fixed by #401.

**Comment by orialb** [comment]
thanks for the quick fix.
